src/app/services/signalr/signalr.service.ts

```diff
@@ -1,45 +1,54 @@
 /**
- * Copyright 2021 Carnegie Mellon University. All Rights Reserved.
+ * Copyright 2023 Carnegie Mellon University. All Rights Reserved.
  * Released under a MIT (SEI)-style license. See LICENSE.md in the project root for license information.
  */
 
 import { Inject, Injectable } from '@angular/core';
 import { ComnAuthService } from '@cmusei/crucible-common';
 import * as signalR from '@microsoft/signalr';
+import { BehaviorSubject, Observable } from 'rxjs';
 import { BASE_PATH } from '../../generated/vm-api';
+import { UserQuery } from '../../state/user/user.query';
 import { UserService } from '../../state/user/user.service';
 import { VsphereService } from '../../state/vsphere/vsphere.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class SignalRService {
+
+  public currentVmUsers$: BehaviorSubject<string[]>;
   private hubConnection: signalR.HubConnection;
   private connectionPromise: Promise<void>;
 
   private userId: string;
   private viewId: string;
+  private vmId: string;
 
   private apiUrl: string;
 
   constructor(
     private authService: ComnAuthService,
     private vmService: VsphereService,
     private userService: UserService,
+    private userQuery: UserQuery,
     @Inject(BASE_PATH) basePath: string
   ) {
     this.apiUrl = basePath;
 
     this.authService.user$.subscribe(() => {
       this.reconnect();
     });
+
+    this.currentVmUsers$ = new BehaviorSubject<string[]>([]);
   }
 
-  public startConnection(): Promise<void> {
+  public startConnection(vmId: string = ''): Promise<void> {
+    this.vmId = vmId;
```

Comment:
Consider setting and unsetting this in setActiveVirtualMachine and unsetActiveVirtualMachine instead of here?

Reply:
Correct, I'm removing this.vmId.  That was necessary for an older version I had working.